function/pkg/operator/operator.go

```diff
@@ -77,6 +78,77 @@ func applyObject(ctx context.Context, c client.Client, u unstructured.Unstructur
 	return response, statusEntryCreated, nil
 }
 
+// updateConfigurationObject copies some elements from the source configuration to the destination configuration.
+// The following items are copied:
+//  * spec,
+//  * metadata/labels,
+//  * metadata/annotations.
+// The rest is unchanged.
+func updateConfigurationObject(destination *unstructured.Unstructured, source unstructured.Unstructured) *unstructured.Unstructured {
+	destination.Object["spec"] = source.Object["spec"]
+
+	destinationMetadata := getMetadataFromConfigurationObject(*destination)
+	sourceMetadata := getMetadataFromConfigurationObject(source)
+
+	updateConfigurationElement(&destinationMetadata, sourceMetadata, "labels")
+	updateConfigurationElement(&destinationMetadata, sourceMetadata, "annotations")
+
+	return destination
+}
+
+func updateConfigurationElement(destination *map[string]interface{}, source map[string]interface{}, elementName string) {
+	sourceElement, ok := source[elementName]
+	if ok {
+		(*destination)[elementName] = sourceElement
+	} else {
+		delete(*destination, elementName)
+	}
+}
+
+// configurationObjectsAreEquivalent compares two structures that represent configuration. Checks that the elements:
+//  * spec,
+//  * metadata/labels,
+//  * metadata/annotations
+// are equal (semantically). The rest are ignored.
```

Comment:
Let's discuss it if we want to add comments for private methods. Maybe a better option would be to add a comment inside the method (if it's not self-explainable)? But I understood this function after a quick look at it :D

Adding comments for methods is a good practice, I can agree. But if I'm not wrong it is valid for public methods which can be imported by other components/services. An example [here](https://github.com/kubernetes/kubectl/blob/4483ba9cf6c54db2163399ba6276173f22cce75f/pkg/cmd/apply/apply.go#L231)